examples/self_encryption/main.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2015 MaidSafe.net limited
+//
+// This MaidSafe Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the MaidSafe Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0, found in the root
+// directory of this project at LICENSE, COPYING and CONTRIBUTOR respectively and also
+// available at: http://www.maidsafe.net/licenses
+//
+// Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied.
+//
+// See the Licences for the specific language governing permissions and limitations relating to
+// use of the MaidSafe
+// Software.
+
+//! Implementation of a simple commandline for Self-Encryption
+
+extern crate self_encryption;
+
+use std::path;
+use std::fs;
+use std::io::*;
+// use tempdir::TempDir as TempDir;
+
+macro_rules! iotry {
+    ($e:expr) => (match $e { Ok(v) => v, Err(e) => panic!("{}", e), })
+}
+
+fn usage() {
+  println!("Usage: self-encrypt or decrypt a local file");
+  println!("       se -e <source_file>");
+  println!("       se -d <source_file>");
+  println!("       note: use absolute paths");
+}
+
+pub struct MyStorage<'b> {
+  source_path : &'b path::Path,
+  name : &'b str;
+  dir : &'b path::Path,
+  destination_path : &'b path::Path
+}
+
+
+impl MyStorage {
+  pub fn new(filepath_ : &str) -> MyStorage {
+    MyStorage { source_path : match path::Path::new(filepath_) {
```

Comment:
I'm really bleeding here... I trying to struggle through but I do not know what I'm doing with Rust and file system :)!